packages/xo-web/src/common/xo/index.js

```diff
@@ -1227,40 +1233,44 @@ export const startVms = vms =>
   }, noop)
 
 export const stopVm = async (vm, force = false) => {
-  try {
-    await confirm({
-      title: _('stopVmModalTitle'),
-      body: _('stopVmModalMessage', { name: vm.name_label }),
+  let forceBlockedOperation = false
+  const id = resolveId(vm)
+  await confirm({
+    title: _('stopVmModalTitle'),
+    body: _('stopVmModalMessage', { name: vm.name_label }),
+  })
+  return _call('vm.stop', { id, force })
+    .catch(async err => {
+      if (operationBlocked.is(err)) {
+        await confirm({
+          title: _('stopVmBlockedModaltitle'),
+          body: _('stopVmBlockedModalMessage'),
+        })
+        forceBlockedOperation = true
+        return _call('vm.stop', { id, force, forceBlockedOperation })
+      }
+      throw err
+    })
+    .catch(async err => {
+      if (err === undefined) {
+        return
+      }
+      if (vmLacksFeature.is(err) || force) {
+        await confirm({
+          title: _('vmHasNoTools'),
+          body: (
+            <div>
+              <p>{_('vmHasNoToolsMessage')}</p>
+              <p>
+                <strong>{_('confirmForceShutdown')}</strong>
+              </p>
+            </div>
+          ),
+        })
+        return _call('vm.stop', { id, force: true, forceBlockedOperation })
```

Comment:
```suggestion
        return _call('vm.stop', { id, force: true, forceBlockedOperation: false })
```

Reply:
This can cause some issues. In case `stop` is a `blocked_operation` this method will throw, even with `force`.
That why i am passing the value of `forceBlockedOperation`. In case it's a `blocked_operation` then it will be `true`, else `false`.